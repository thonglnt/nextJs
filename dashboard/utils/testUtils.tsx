import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
