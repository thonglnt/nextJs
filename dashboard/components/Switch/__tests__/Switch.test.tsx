// Utils
import { render } from '@utils/testUtils'

// Components
import SwitchLabels, { ISwitchLabels } from '..'

describe('Switch component', () => {
  const mockProps: ISwitchLabels = {
    label: 'Switch',
    defaultChecked: true,
  }

  it('should render Switch component', () => {
    const { container } = render(<SwitchLabels {...mockProps} />)

    expect(container).toMatchSnapshot()
  })
})
