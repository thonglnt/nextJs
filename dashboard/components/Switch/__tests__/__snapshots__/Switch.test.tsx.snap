// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component should render Switch component 1`] = `
<div>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
  >
    <span
      class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
    >
      <span
        class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary Mui-checked css-12f33q6-MuiButtonBase-root-MuiSwitch-switchBase"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
          type="checkbox"
        />
        <span
          class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
      />
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
    >
      Switch
    </span>
  </label>
</div>
`;
